core/src/main/java/tc/oc/pgm/flag/state/Uncarried.java

```diff
@@ -142,24 +107,14 @@ protected boolean inPickupRange(Player player) {
     return false;
   }
 
-  protected boolean canPickup(MatchPlayer player) {
-    if (this.pickingUp != null) return false; // Prevent infinite recursion
-
-    for (Flag flag : this.flag.getMatch().getModule(FlagMatchModule.class).getFlags()) {
-      if (flag.isCarrying(player)) return false;
-    }
-
-    return this.flag.canPickup(player, this.post);
-  }
-
   @Override
   public void onEvent(PlayerMoveEvent event) {
     super.onEvent(event);
     MatchPlayer player = this.flag.getMatch().getPlayer(event.getPlayer());
     if (player == null || !player.canInteract() || player.getBukkit().isDead()) return;
 
     if (this.inPickupRange(player.getBukkit()) && this.canPickup(player)) {
-      this.pickupFlag(player);
+      this.flag.pickupFlag(player, this.location);
```

Comment:
I wouldnt route this thru flag, you have the method directly in state
```suggestion
      this.pickupFlag(player, this.location);
```